### PR TITLE
feat(auth): tenant service

### PR DESCRIPTION
**Pull request by BlairCurrey**
## Changes proposed in this pull request

-  Adds tenant service to `auth`

## Context

- Fixes https://github.com/interledger/rafiki/issues/3122

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [x] Make sure that all checks pass
- [x] Bruno collection updated (if necessary)
- [x] Documentation issue created with `user-docs` label (if necessary)
- [x] OpenAPI specs updated (if necessary)

### CODE REVIEW

File: packages/auth/src/tenant/service.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    tenantService = await deps.use('tenantService')
+
+    // TODO: remove. temporary implementation while other issue is completed
+    const knex = await deps.use('knex')
+    await knex.schema.createTable('tenants', (table) => {
+      table.string('id').primary()
+      table.string('idpConsentUrl').notNullable()
+      table.string('idpSecret').notNullable()
+    })
+  })
```
Comment (BlairCurrey) [inline packages/auth/src/tenant/service.test.ts:30]: Need to remove - patched in while waiting on actual migration to get merged in.

File: packages/auth/src/tenant/service.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { TransactionOrKnex, Model } from 'objection'
+
+// TODO: remove. temporary implementation while other issue is completed
+export class Tenant extends Model {
+  static tableName = 'tenants'
+
+  id!: string
+  idpConsentUrl!: string
+  idpSecret!: string
+}
```
Comment (BlairCurrey) [inline packages/auth/src/tenant/service.ts:11]: Need to remove - patched in while waiting on actual migration to get merged in.

File: packages/auth/src/tenant/service.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  input: Partial<Omit<CreateOptions, 'id'>>
+): Promise<Tenant | undefined> {
+  return await Tenant.query(deps.knex).patchAndFetchById(id, input)
+}
+
+async function deleteTenant(
+  deps: ServiceDependencies,
+  id: string
+): Promise<boolean> {
+  const deleted = await Tenant.query(deps.knex).deleteById(id)
```
Comment (mkurapov) [inline packages/auth/src/tenant/service.ts:72]: lets do a soft delete here, we can already leverage the `deletedAt`

Comment (BlairCurrey) [reply packages/auth/src/tenant/service.ts]: updated to soft delete

**Review by mkurapov — APPROVED** [review]

**Review by sanducb — APPROVED** [review]

**Pull request merged**